apps/04TowersOfHanoi.js

```diff
@@ -19,31 +19,51 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+    var block = stacks[startStack].pop();
+    var towerEnd = stacks[endStack].push(block);
 }
 
-function isLegal() {
-  // Your code here
+function isLegal(startStack, endStack) {
+
+  var block2 = stacks[startStack][stacks[startStack].length -1];
+  var end = stacks[endStack][stacks[endStack].length -1];
+
+  if (stacks[endStack].length === 0) {
+      return true;
+  }
 
+  if (block2 < end) {
```

Comment:
Try simply using a 'block' variable here, so ppl don't get stuck looking for block1, block3, etc. varaiables in your app.